include/boost/thread/pthread/timespec.hpp

```diff
@@ -278,20 +303,52 @@ namespace boost
     static inline mono_timespec_timepoint now()
     {
 #if defined(BOOST_THREAD_PLATFORM_WIN32)
-      win32::tick_types msec = win32::GetTickCount64_()();
+#if defined(BOOST_THREAD_USES_CHRONO)
+      // Use QueryPerformanceCounter() to match the implementation in Boost
+      // Chrono so that chrono::steady_clock::now() and this function share the
+      // same epoch and so can be converted between each other.
+      boost::detail::winapi::LARGE_INTEGER_ freq;
+      if ( !boost::detail::winapi::QueryPerformanceFrequency( &freq ) )
+      {
+        BOOST_ASSERT(0 && "Boost::Thread - QueryPerformanceFrequency Internal Error");
+        return mono_timespec_timepoint(0);
+      }
+      if ( freq.QuadPart <= 0 )
+      {
+        BOOST_ASSERT(0 && "Boost::Thread - QueryPerformanceFrequency Internal Error");
+        return mono_timespec_timepoint(0);
+      }
+
+      boost::detail::winapi::LARGE_INTEGER_ pcount;
+      unsigned times=0;
+      while ( ! boost::detail::winapi::QueryPerformanceCounter( &pcount ) )
+      {
+        if ( ++times > 3 )
+        {
+          BOOST_ASSERT(0 && "Boost::Thread - QueryPerformanceCounter Internal Error");
+          return mono_timespec_timepoint(0);
+        }
+      }
+
+      long double ns = 1000000000.0L * pcount.QuadPart / freq.QuadPart;
+      return mono_timespec_timepoint(static_cast<boost::intmax_t>(ns));
+#else
+      // Use GetTickCount64() because it's more reliable on older
```

Comment:
Does it means that Boost.Chrono should use this on those systems?

Reply:
Yes, that is correct. Boost Chrono should probably provide an option so users can use GetTickCount64() instead of QueryPerformanceCounter() when using chrono::steady_time on older systems. Right now the only option those users have is to avoid using chrono::steady_time on those systems.

Reply:
Issue added https://github.com/boostorg/chrono/issues/28

Reply:
That would be great! To be exact, it's the GetTickCount64 emulation that works best on those systems. Unfortunately, GetTickCount64 itself has also only been introduced in Windows Vista.